src/Mod/Arch/ArchWall.py

```diff
@@ -421,7 +432,12 @@ def addDefault(self):
 
         FreeCADGui.addModule("Draft")
         if FreeCAD.ParamGet("User parameter:BaseApp/Preferences/Mod/Arch").GetBool("WallSketches",True):
-            FreeCADGui.doCommand('base=FreeCAD.ActiveDocument.addObject("Sketcher::SketchObject","WallTrace")')
+            # Use ArchSketch if SketchArch add-on is present
+            try:
+                FreeCADGui.doCommand('import ArchSketchObject')
```

Comment:
I would here do the `import ArchSketchObject` outside of the FreeCADGui.doCommand, because it's an internal test, it might give confusing info to the user if it fails

Reply:
Thanks for the comment - in fact I have no idea what exactly is 'doCommand' - will do as suggested :)

Reply:
Thanks, tested the revised code and seems working; would upload to the forum for testings by other users.

Reply:
I would do this because then if the import fails, the user does not see the error message

```suggestion
                import ArchSketchObject
                FreeCADGui.doCommand('import ArchSketchObject')
```

Reply:
Thanks for the comments  -  would do together with the 'dummy' ArchSketchObject.py to avoid failure of ArchSketch object if the external add-on is not installed.

Wondering what is the difference between the 2 lines of code which looks similar.

Reply:
they both do the same thing, but using the doCommand, it first prints the string to the Python console, then executes the line. So the user sees the import line being printed. But it's rather cosmetic anyway.. Let's merge now and we can fix that later on. Is it OK to merge?

Reply:
Yes please :)

Reply:
Thanks!